### PR TITLE
Improve coverage.sh usability when used locally

### PR DESCRIPTION
**Pull request by ryoqun**
#### Summary of Changes

- Restrict cleared coverage files really to be regular files for responsibility separation for each code blocks
- Guard on-going `find` invocation with `cov/after-test` against newly-created symlinks to be matched
- Always generate report and restore non-successful exit status at last if any
- Introduce `LATEST` symlink for easier reloading without re-navigating.
- Skip full-clean build when not under CI or when testing only some.

### CODE REVIEW

File: scripts/coverage.sh [hunk 1]
```diff
@@ -31,27 +31,35 @@ export RUST_MIN_STACK=8388608
...
+  # shellcheck disable=SC2046
+  touch \
+    $(git ls-files :**/build.rs) \
+    $(git grep -l "proc-macro.*true" :**/Cargo.toml | sed 's|Cargo.toml|src/lib.rs|')
+fi
 
 RUST_LOG=solana=trace _ cargo +$rust_nightly test --target-dir target/cov --no-run "${packages[@]}"
-RUST_LOG=solana=trace _ cargo +$rust_nightly test --target-dir target/cov "${packages[@]}" 2> target/cov/coverage-stderr.log
+if RUST_LOG=solana=trace _ cargo +$rust_nightly test --target-dir target/cov "${packages[@]}" 2> target/cov/coverage-stderr.log;
+then
```
Comment (mvines) [inline scripts/coverage.sh]:
```suggestion
if RUST_LOG=solana=trace _ cargo +$rust_nightly test --target-dir target/cov "${packages[@]}" 2> target/cov/coverage-stderr.log; then
```

File: scripts/coverage.sh [hunk 2]
```diff
@@ -115,3 +123,9 @@ genhtml --output-directory target/cov/$reportName \
 )
 
 ls -l target/cov/$reportName/index.html
+ln -sfT $reportName target/cov/LATEST
+
+if [[ -n $failure_code ]]
+then
+  exit $failure_code
+fi
```
Comment (mvines) [inline scripts/coverage.sh]: Set `failure_code=0` on line 58 and then `exit $failure_code` here unconditionally?

Comment (ryoqun) [reply scripts/coverage.sh]: nice!

File: scripts/coverage.sh (hunk 1 above)
Comment (ryoqun) [reply scripts/coverage.sh]: Uh, oh! My style slipped in. :p

**Comment by ryoqun** [comment]
I wanted this for #9357, so back-port to v1.0. But, I hope we don't need to debug that branch in the future....

**Pull request merged**